src/servicebus/azext_servicebus/_params.py

```diff
@@ -0,0 +1,357 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+from azure.cli.core.commands.parameters import (tags_type, get_enum_type, resource_group_name_type, name_type)
+
+from azext_servicebus._validators import _validate_auto_delete_on_idle, \
+    _validate_duplicate_detection_history_time_window, _validate_default_message_time_to_live, _validate_lock_duration
+
+
+# pylint: disable=line-too-long
+def load_arguments_namespace(self, _):
+    with self.argument_context('servicebus namespace exists') as c:
+        c.argument('namespace_name', options_list=['--name'], help='name of the Namespace')
+
+    with self.argument_context('servicebus namespace create') as c:
+        c.argument('resource_group_name', arg_type=resource_group_name_type)
+        c.argument('namespace_name', options_list=['--name'], help='name of the Namespace')
+        c.argument('tags', options_list=['--tags', '-t'], arg_type=tags_type, help='tags for the namespace in '
+                                                                                   'Key value pair format')
+        c.argument('sku', options_list=['--sku-name'], arg_type=get_enum_type(['Basic', 'Standard', 'Premium']))
+        c.argument('location', options_list=['--location', '-l'], help='Location')
+        c.argument('skutier', options_list=['--sku-tier'], arg_type=get_enum_type(['Basic', 'Standard', 'Premium']))
+        c.argument('capacity', options_list=['--capacity'], help='Capacity for Sku')
+
+    # region Namespace Get
+    for scope in ['servicebus namespace show', 'servicebus namespace delete']:
+        with self.argument_context(scope) as c:
+            c.argument('resource_group_name', arg_type=resource_group_name_type)
+            c.argument('namespace_name', options_list=['--name', '-n'], help='name of the Namespace')
+
+    with self.argument_context('servicebus namespace list') as c:
+        c.argument('resource_group_name', arg_type=resource_group_name_type)
```

Comment:
Do this at the start of _params.py.

Then you don't have to keep on doing it for each command as it will apply for all.
Then you can remove all the `c.argument('resource_group_name', arg_type=resource_group_name_type,)` everywhere.
```
with self.argument_context('servicebus') as c:
    c.argument('resource_group_name', arg_type=resource_group_name_type)
```

Reply:
Resolved,  please verify .